### ISSUE TITLE
doctest and exUnit lineendings issue on windows

### ISSUE THREAD

**Issue by DeemoONeill** [elixir-lang/elixir#12826]
### Elixir and Erlang/OTP versions

Erlang/OTP 25 [erts-13.0.4] [source] [64-bit] [smp:6:6] [ds:6:6:10] [async-threads:1] [jit:ns]

Elixir 1.14.0 (compiled with Erlang/OTP 25)

### Operating system

Windows

### Current behavior

Doctest Minimal example:

```elixir
defmodule DoctestIssue do
  @moduledoc """
  Documentation for `DoctestIssue`.
  """

  @doc """
  Hello world.
  
  ## Examples
  
      iex> DoctestIssue.hello()
      "hello
      world"
  
  """
  def hello do
    "hello
world"
  end
end
```

ExUnit example:

```elixir
defmodule DoctestIssueTest do
  use ExUnit.Case
  doctest DoctestIssue

  test "greets the world" do
    assert DoctestIssue.hello() == "hello
world"
  end
end
```

`mix test`

```elixir
  1) doctest DoctestIssue.hello/0 (1) (DoctestIssueTest)
     test/doctest_issue_test.exs:3
     Doctest failed
     doctest:
       iex> DoctestIssue.hello()
       "hello
       world"
     code:  DoctestIssue.hello() === "hello
            world"
     left:  "hello\r\nworld"
     right: "hello\nworld"
     stacktrace:
       lib/doctest_issue.ex:11: DoctestIssue (module)

 2) test greets the world (DoctestIssueTest)
     test/doctest_issue_test.exs:5
     Assertion with == failed
     code:  assert DoctestIssue.hello() == "hello\nworld"
     left:  "hello\r\nworld"
     right: "hello\nworld"
     stacktrace:
       test/doctest_issue_test.exs:6: (test)
```

### Expected behavior

The issue seems to be that default line endings on windows is `\r\n`. 
Any code written in the doctest with `\r\n` lineendings is converted to `\n`. 

I first noticed this issue while running the test suite for explorer and having failing tests after I had saved a file (and my editor converted line endings on the saved file).

If I set my editor to use LF line endings then the issue resolves, But CLRF causes failures for multiline strings. 

looking at doctest it looks like `\n` is the assumed lineending, is there a way to check like `String.ends_with?("\r\n")` and pass the lineendings down to the functions that construct the test? 

I'm also happy to put my hands up and configure my editor to use LF lineendings

**Comment by josevalim** [comment]
Thanks for the report. I will take a look. :) We already have some \r\n tests but not within the string.

**Comment by DeemoONeill** [comment]
The ones that got me in explorer are the dataframe table tests, 

     left:  "+-----------------------------------------------------------------------+\n|              Explorer DataFrame: [rows: 150, columns: 5]              |\n+--------------+-------------+--------------+-------------+-------------+\n| sepal_length | sepal_width | petal_length | petal_width |   species   |\n|   <float>    |   <float>   |   <float>    |   <float>   |  <string>   |\n+==============+=============+==============+=============+=============+\n| 5.1          | 3.5         | 1.4          | 0.2         | Iris-setosa |\n+--------------+-------------+--------------+-------------+-------------+\n\n"
     right: "+-----------------------------------------------------------------------+\r\n|              Explorer DataFrame: [rows: 150, columns: 5]              |\r\n+--------------+-------------+--------------+-------------+-------------+\r\n| sepal_length | sepal_width | petal_length | petal_width |   species   |\r\n|   <float>    |   <float>   |   <float>    |   <float>   |  <string>   |\r\n+==============+=============+==============+=============+=============+\r\n| 5.1          | 3.5         | 1.4          | 0.2         | Iris-setosa |\r\n+--------------+-------------+--------------+-------------+-------------+\r\n\r\n"

**Comment by josevalim** [comment]
Hi @DeemoONeill, I could not reproduce this on Elixir v1.15. We rewrote the doctests parser for v1.15 and it seems it has addressed this issue. Can you reproduce this on latest? :)

**Comment by DeemoONeill** [comment]
I'm on holiday at the moment. I'm back on September 1st I'll give it a go then :)

**Comment by josevalim** [comment]
Btw, there is something else to pay attention too, your second test also fails:

```
 2) test greets the world (DoctestIssueTest)
     test/doctest_issue_test.exs:5
     Assertion with == failed
     code:  assert DoctestIssue.hello() == "hello\nworld"
     left:  "hello\r\nworld"
     right: "hello\nworld"
     stacktrace:
       test/doctest_issue_test.exs:6: (test)
```

But there are no doctests here. So could it be that one file had CRLF and the other LF?

**Comment by josevalim** [comment]
Enjoy the holidays!

**Comment by DeemoONeill** [comment]
Apologies for the delay, the issue does seem to be resolved on version 1.15.5. Thanks!
